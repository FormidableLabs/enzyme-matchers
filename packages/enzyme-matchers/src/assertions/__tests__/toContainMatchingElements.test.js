const PropTypes = require('prop-types');

const toContainMatchingElements = require('../toContainMatchingElements');

function User(props) {
  return (
    <span className={props.className}>
      User {props.index}
    </span>
  );
}

User.propTypes = {
  index: PropTypes.number.isRequired,
  className: PropTypes.string,
};

function Fixture() {
  return (
    <div>
      <ul>
        <li>
          <User index={1} className="userOne" />
        </li>
        <li>
          <User index={2} className="userTwo" />
        </li>
      </ul>
    </div>
  );
}

describe('toContainMatchingElements', () => {
  const wrapper = shallow(<Fixture />);
  const truthyResults = [
    toContainMatchingElements(wrapper, 2, 'User'),
    toContainMatchingElements(wrapper, 1, '[index=1]'),
    toContainMatchingElements(wrapper, 2, '[index]'),
    toContainMatchingElements(wrapper.find('ul'), 2, '[index]'),
    toContainMatchingElements(wrapper.find('User'), 1, '[index=1]'),
  ];
  const falsyResults = [
    toContainMatchingElements(wrapper, 3, 'User'),
    toContainMatchingElements(wrapper, 1, '.userThree'),
    toContainMatchingElements(wrapper, 1, '[index]'),
    toContainMatchingElements(wrapper.find('ul'), 3, '[index]'),
  ];

  it('returns the pass flag properly', () => {
    truthyResults.forEach(truthyResult => {
      expect(truthyResult.pass).toBe(true);
    });
    falsyResults.forEach(falsyResult => {
      expect(falsyResult.pass).toBe(false);
    });
  });

  it('returns the message with the proper pass verbage', () => {
    truthyResults.forEach(truthyResult => {
      expect(truthyResult.message).toMatchSnapshot();
    });
  });

  it('returns the message with the proper fail verbage', () => {
    falsyResults.forEach(falsyResult => {
      expect(falsyResult.negatedMessage).toMatchSnapshot();
    });
  });

  it('provides contextual information for the message', () => {
    truthyResults.forEach(truthyResult => {
      expect(truthyResult.contextualInformation).toMatchSnapshot();
    });
  });
});
