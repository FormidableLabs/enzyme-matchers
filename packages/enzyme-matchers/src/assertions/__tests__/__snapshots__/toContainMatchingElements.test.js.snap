// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toContainMatchingElements provides contextual information for the message 1`] = `
Object {
  "actual": "HTML Output of <div>:
 <div><ul><li><User index={1} className=\\"userOne\\" /></li><li><User index={2} className=\\"userTwo\\" /></li></ul></div>",
}
`;

exports[`toContainMatchingElements provides contextual information for the message 2`] = `
Object {
  "actual": "HTML Output of <div>:
 <div><ul><li><User index={1} className=\\"userOne\\" /></li><li><User index={2} className=\\"userTwo\\" /></li></ul></div>",
}
`;

exports[`toContainMatchingElements provides contextual information for the message 3`] = `
Object {
  "actual": "HTML Output of <div>:
 <div><ul><li><User index={1} className=\\"userOne\\" /></li><li><User index={2} className=\\"userTwo\\" /></li></ul></div>",
}
`;

exports[`toContainMatchingElements returns the message with the proper fail verbage 1`] = `"Expected <div> to not contain 3 elements matching User but it did."`;

exports[`toContainMatchingElements returns the message with the proper fail verbage 2`] = `"Expected <div> to not contain 1 element matching .userThree but it did."`;

exports[`toContainMatchingElements returns the message with the proper fail verbage 3`] = `"Expected <div> to not contain 1 element matching [index] but it did."`;

exports[`toContainMatchingElements returns the message with the proper pass verbage 1`] = `"Expected <div> to contain 2 elements matching User but 2 were found."`;

exports[`toContainMatchingElements returns the message with the proper pass verbage 2`] = `"Expected <div> to contain 1 element matching [index=1] but 1 was found."`;

exports[`toContainMatchingElements returns the message with the proper pass verbage 3`] = `"Expected <div> to contain 2 elements matching [index] but 2 were found."`;
